NetKet/pynetket.cc

```diff
@@ -42,17 +42,29 @@ PYBIND11_MODULE(pynetket, m) {
   py::bind_vector<std::vector<std::vector<double>>>(m, "VectorVectorDouble");
   py::bind_vector<std::vector<std::complex<double>>>(m, "VectorComplexDouble");
 
-  py::class_<Graph>(m, "Graph")
-      .def(py::init<std::string, py::kwargs>())
-      .def("Nsites", &Graph::Nsites)
-      .def("AdjacencyList", &Graph::AdjacencyList)
-      .def("SymmetryTable", &Graph::SymmetryTable)
-      .def("EdgeColors", &Graph::EdgeColors)
-      .def("IsBipartite", &Graph::IsBipartite)
-      .def("IsConnected", &Graph::IsConnected)
-      .def("Distances", &Graph::Distances)
-      .def("AllDistances", &Graph::AllDistances);
+  py::class_<AbstractGraph>(m, "_AbstractGraph")
+      .def("Nsites", &AbstractGraph::Nsites)
+      .def("AdjacencyList", &AbstractGraph::AdjacencyList)
+      .def("SymmetryTable", &AbstractGraph::SymmetryTable)
+      .def("EdgeColors", &AbstractGraph::EdgeColors)
+      .def("IsBipartite", &AbstractGraph::IsBipartite)
+      .def("IsConnected", &AbstractGraph::IsConnected)
+      .def("Distances", &AbstractGraph::Distances)
+      .def("AllDistances", &AbstractGraph::AllDistances);
 
+  py::class_<Hypercube, AbstractGraph>(m, "Hypercube")
+      .def(py::init<int, int, bool>())
```

Comment:
We should make sure to use [keyword arguments](https://pybind11.readthedocs.io/en/stable/basics.html#keyword-arguments) here, so that everything is transparent from the user's perspective! (small change, but worth doing it from the beginning I guess)